js/Common.js

```diff
@@ -1638,8 +1648,28 @@ if (!CRM.vars) CRM.vars = {};
 
   // Format a date for output to the user
   // Input may be a js Date object, a unix timestamp or a yyyy-mm-dd string
-  CRM.utils.formatDate = function(input, outputFormat) {
-    return input ? $.datepicker.formatDate(outputFormat || CRM.config.dateInputFormat, CRM.utils.makeDate(input)) : '';
+  CRM.utils.formatDate = function(input, outputFormat, showTime) {
+    if (!input) {
+      return '';
+    }
+    var date = CRM.utils.makeDate(input),
+      output = $.datepicker.formatDate(outputFormat || CRM.config.dateInputFormat, date);
+    if (showTime) {
+      var hour = date.getHours(),
+        min = date.getMinutes(),
+        suf = '';
+      if (showTime === 12 || (showTime === true && !CRM.config.timeIs24Hr)) {
+        suf = ' ' + ts('AM');
+        if (hour === 0) {
+          hour = 12;
+        } else if (hour > 12) {
+          hour -= 12;
+          suf = ' ' + ts('PM');
+        }
+      }
+      output += ' ' + hour + ':' + (min < 10 ? '0' : '') + min + suf;
```

Comment:
I'm not familiar with all locales but do they all use ':' as time separators?

Reply:
Dunno. Patches welcome I guess :)

Reply:
My thinking here is this block is neither locale-aware nor uses the time format configuration in civi settings, and I was kind of surprised civi doesn't already have a library for js dates since this must have been something people have dealt with hundreds of times already. I realize it's not trivial to just add one, partly because there'll be differences between supported format strings/locale names/etc.

Having said that, I suppose this would have come up elsewhere if people were really concerned about time formats. 12/24 hour is likely all they really care about.

Could you add a param declaration describing what showTime is? And also maybe for outputFormat, indicating only the date part is supported?